src/cplscheme/SerialCouplingScheme.cpp

```diff
@@ -108,26 +111,46 @@ void SerialCouplingScheme::exchangeFirstData()
     sendData(getM2N(), getSendData());
   } else { // second participant
     if (isImplicitCouplingScheme()) {
-      PRECICE_DEBUG("Test Convergence and accelerate...");
+      PRECICE_DEBUG("Perform acceleration (only second participant)...");
       doImplicitStep();
+    }
+  }
+
+  // @todo bundle sendConvergence, specific moveToNextWindow and sendData into one function?
+  if (not doesFirstStep()) {
+    if (isImplicitCouplingScheme()) {
+      PRECICE_DEBUG("Sending convergence...");
       sendConvergence(getM2N());
     }
+    if (hasConverged() || isExplicitCouplingScheme()) {
+      moveToNextWindow(); // do moveToNextWindow already here for second participant in SerialCouplingScheme
+    }
     PRECICE_DEBUG("Sending data...");
     sendData(getM2N(), getSendData());
   }
 }
 
 void SerialCouplingScheme::exchangeSecondData()
 {
+  // @todo bundle receiveConvergence, specific moveToNextWindow and sendData into one function?
   if (doesFirstStep()) { // first participant
-    PRECICE_DEBUG("Receiving convergence data...");
     if (isImplicitCouplingScheme()) {
+      PRECICE_DEBUG("Receiving convergence data...");
       receiveConvergence(getM2N());
     }
+    if (hasConverged() || isExplicitCouplingScheme()) {
+      moveToNextWindow(); // extrapolation result for receive data of first is directly overwritten in next step
```

Comment:
To what do you refer with "next step" here?

Reply:
That was unclear. Changed it in 600e855b290b0e95e884812000cdf05a35fe41e4